src/app/leverage/provider.tsx

```diff
@@ -239,6 +245,47 @@ export function LeverageProvider(props: { children: any }) {
     fetchStats()
   }, [baseToken])
 
+  useEffect(() => {
+    async function fetchCostOfCarry(
+      jsonRpcProvider: providers.JsonRpcProvider,
+    ) {
+      const poolDataProviderContract = new UiPoolDataProvider({
+        uiPoolDataProviderAddress: AaveV3Ethereum.UI_POOL_DATA_PROVIDER,
+        provider: jsonRpcProvider,
+        chainId: ChainId.mainnet,
+      })
+      const reserves = await poolDataProviderContract.getReservesHumanized({
+        lendingPoolAddressProvider: AaveV3Ethereum.POOL_ADDRESSES_PROVIDER,
+      })
+
+      const formattedPoolReserves = formatReserves({
+        reserves: reserves.reservesData,
+        currentTimestamp: Math.floor(Date.now() / 1000),
+        marketReferenceCurrencyDecimals:
+          reserves.baseCurrencyData.marketReferenceCurrencyDecimals,
+        marketReferencePriceInUsd:
+          reserves.baseCurrencyData.marketReferenceCurrencyPriceInUsd,
+      })
```

Comment:
Logic is from https://github.com/aave/aave-utilities?tab=readme-ov-file#formatreserves